### PR TITLE
Modify to use templates

### PR DESCRIPTION
**Pull request by finanalyst**
This is a partial modification to move inline HTML statements into templates.
- Currently only one template is used, but it significantly cleans up the perl6 code for pod2html
- The templating module is Template::Mustache. Although Mojo was my first choice, it is not currently passing all tests, whilst Template::Mustache is passing all tests.
- A template allows a user to provide custom HTML.
- A 'partal' modification because various other routines generate little bits of HTML. Some, perhaps do not need to be changed because they just wrap text into obvious tags.
- The toc subroutine output should also be moved to a template. TODO.
- Added tests for templates. Since the lack of templates generates output on stderr, I have included Test::Output to capture and verify the error message in the test code.

### CODE REVIEW

File: t/templates/main.mustache [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+<html lang="{{ lang }}">
+    <head>
+        <title>{{ title }}</title>
+        <meta charset="UTF-8" />
+        <meta description="This is a new template"/>
+        <style>
+        /* code gets the browser-default font
+         * kbd gets a slightly less common monospace font
+         * samp gets the hard pixelly fonts
+         */
```
Comment (zoffixznet) [inline t/templates/main.mustache]: I don't think we need these comments up in there. Some sites can be high-traffic ones (like docs.perl6.org) and there's no reason for them to pay for extra bandwidth due to some fairly useless comments in markup.

File: lib/Pod/To/HTML.pm [hunk 2]
```diff
@@ -1,5 +1,6 @@
 unit class Pod::To::HTML;
 use URI::Escape;
+use Template::Mustache;
```
Comment (zoffixznet) [inline lib/Pod/To/HTML.pm]: Doesn't look like `Template::Mustache` was added to `"depends"` section in `META6.json`

**Comment by zoffixznet** [comment]
> finanalyst requested your review on this pull request.

I left a couple of comments, but I think someone from the docs team could be a better judge of this work. I see @JJ opened https://github.com/perl6/Pod-To-HTML/issues/35 , perhaps he could review this PR as well.

File: lib/Pod/To/HTML.pm (hunk 2 above)
Comment (finanalyst) [reply lib/Pod/To/HTML.pm]: Fixing. Travis also complained.

File: lib/Pod/To/HTML.pm [hunk 3]
```diff
@@ -157,17 +158,48 @@ sub assemble-list-items(:@content, :$node, *% ) {
...
-  :$css-url = '//design.perl6.org/perl.css', :$lang = 'en',
+sub pod2html(
+    $pod,
+    :&url = -> $url { $url },
+    :$head = '',
+    :$header = '',
+    :$footer = '',
+    :$default-title,
+    :$css-url = '//design.perl6.org/perl.css',
+    :$templates,
```
Comment (JJ) [inline lib/Pod/To/HTML.pm]: There should probably be a sensible default for this; in general, interface changes should be minimized, or we'll deprecate all of a sudden everything that used this.

File: lib/Pod/To/HTML.pm [hunk 4]
```diff
@@ -157,17 +158,48 @@ sub assemble-list-items(:@content, :$node, *% ) {
...
+    :$lang = 'en'
     --> Str ) is export {
+
+    my $template-dir = 'resources/templates';
```
Comment (JJ) [inline lib/Pod/To/HTML.pm]: And this should probably be an argument, with this value as a default. Don't like too much hardcoded paths (or mostly anything).

File: lib/Pod/To/HTML.pm [hunk 5]
```diff
@@ -157,17 +158,48 @@ sub assemble-list-items(:@content, :$node, *% ) {
...
+        :meta( do-metadata ),
+        :$head,
+        :toc( do-toc($pod) ),
+        :$header,
+        :@body,
+        :footnotes( do-footnotes ),
+        :$footer).hash
+    );
+
+=comment out
```
Comment (JJ) [inline lib/Pod/To/HTML.pm]: As long as it's passing tests, it's probably better to just eliminate old code.

File: t/templates/main.mustache (hunk 1 above)
Comment (JJ) [reply t/templates/main.mustache]: But this is a test file. I don't think it matters too much.

Comment (zoffixznet) [reply t/templates/main.mustache]: That comment was meant to go to the main template and half the comments were already removed https://github.com/perl6/Pod-To-HTML/pull/47/commits/5259257f898f07331dae48021187e7a1ec787530

**Comment by JJ** [comment]
Can you please take a look at these changes? We can then merge it.

File: lib/Pod/To/HTML.pm (hunk 4 above)
Comment (finanalyst) [reply lib/Pod/To/HTML.pm]: This is actually wrong. It works in a test environment in a module, but it does not work when 'wild'. The resource needs to be in referenced in META6.json and then in %?RESOURCES. I will change.

**Pull request merged**